README.md

```diff
@@ -1,18 +1,94 @@
-# pytest-unordered
+# pytest-unordered: Test collection content, ignoring order
 
 [![Build Status](https://travis-ci.org/utapyngo/pytest-unordered.svg?branch=master)](https://travis-ci.org/utapyngo/pytest-unordered)
 [![Coverage Status](https://codecov.io/gh/utapyngo/pytest-unordered/branch/master/graph/badge.svg)](https://codecov.io/gh/utapyngo/pytest-unordered)
 
-Test equality of unordered collections in pytest.
+`pytest_unordered` allows you to write simple (pytest) assertions
+to test whether collections have the same content, regardless of order.
+For example:
+
+    assert [1, 20, 300] == unordered([20, 300, 1])
+
 
 ## Installation
 
     pip install pytest-unordered
-    
+
 
 ## Usage
 
+### Basics
+
+In most cases you just need the `unordered()` helper function:
+
     from pytest_unordered import unordered
-    
-    def test_unordered():
-        assert [3, 2, {1: ['b', 'a']}] == unordered({1: unordered('a', 'b')}, 2, 3)
+
+Compare list or tuples by wrapping your expected value with `unordered()`:
+
+    assert [1, 20, 300] == unordered([20, 300, 1])  # Pass
+    assert (1, 20, 300) == unordered((20, 300, 1))  # Pass
+
+Excessive/missing items will be reported by pytest:
+
+    assert [1, 20, 300] == unordered([20, 300, 1, 300])
+
+      E         Extra items in the right sequence:
+      E         300
+
+By default, the container type has to match too:
+
+    assert (1, 20, 300) == unordered([20, 300, 1])
+
+      E         Type mismatch:
+      E         <class 'tuple'> != <class 'list'>
+
+
+
+### Nesting
+
+A seasoned developer will notice that the simple use cases above
+can be easily be addressed with appropriate usage
+of builtins like `set`, `sorted` and `isinstance`.
```

Comment:
`set` and `sorted` do not work with dictionaries.
`set([{'a': 2}, {'a': 1}]) == {{'a': 1}, {'a': 2}}` gives a TypeError, while `unordered([{'a': 2}, {'a': 1}]) == [{'a': 1}, {'a': 2}]` return True.

Reply:
yes, indeed, good point. I'll try to include that use case